### ISSUE TITLE
Unable to upgrade cluster version due to APM issue

### ISSUE THREAD

**Issue by koushikgongireddy** [elastic/terraform-provider-ec#873]
## Readiness Checklist

- [ ] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
Terraform apply should return success.

## Current Behavior
Trying to update Elastic Cluster from 7.17.6 to 7.17.24 and am seeing issues with apm, below is the Terraform error

╷
│ Error: failed updating deployment: 1 error occurred:
│   * api error: deployments.topology_element_using_legacy_ic: The referenced instance configuration [gcp.apm.1] in topology element [<unknown>] is no longer available .
│ 
│ 
│ 
│   with module.ec.ec_deployment.cluster,
│   on ../../gcp-modules/elastic-cloud/cluster.tf line 3, in resource "ec_deployment" "cluster":
│    3: resource "ec_deployment" "cluster" {
│ 

## Terraform definition
We are not specifying anything for APM and directly reading what provider offers
example
```
resource "ec_deployment" "cluster" {
  # Optional name.
  name = "${var.environment}-${var.cluster_name_suffix}${var.hash_suffix}"

  # Mandatory fields
  region                 = "gcp-${var.location}"
  version                = var.es_version
   
   apm {}
}


## Context
1. Terragrunt apply


## Possible Solution
Need to apply TF changes without any issues when there is no change in TF version or elastic version

## Your Environment
<!--- Include as many relevant details about the environment -->
<!--- you experienced the bug in. -->
* Version used:
```
terraform {
  required_version = "1.5.5"

  required_providers {
    google = "5.12.0"
    ec = {
      source  = "elastic/ec"
      version = "0.5.1"
    }
  }
}
```

**Comment by koushikgongireddy** [comment]
@tobio  Can you please check on this issue!!
We are facing issue with elastic cluster upgrade from 7.17.6 to 7.17.24

**Comment by tobio** [comment]
> Does that means we need to remove it from our config and removing it will cause any issues on the cluster?

Nope, continuing to use APM won't cause issues with you deployment. APM has been essentially replaced by the `integrations_server` component in deployments running Elastic Stack 8+, but will continue to be supported in 7.x deployments. 

> Also we are using gcp-io-optimized for deployment_template_id is that causing this issue?

Looking at the error, I believe this is causing the issue. I'd expect migrating to a current deployment template will rectify the problem for you. `gcp-general-purpose` looks like the most similar option. 

Changing the deployment template may cause changes to the allocated CPU/storage resources, and the hourly cost. You can view the potential changes in the Cloud console, however we don't surface this data in the provider at the moment. 

Hit the `Edit` button on the current hardware profile in the deployment overview:
![Image](https://github.com/user-attachments/assets/b274e4a2-43ad-4d27-a87e-35f51d66cb1d)

Then when you select a new hardware profile, you'll see a breakdown of any changes that apply:
![Image](https://github.com/user-attachments/assets/52ffd315-415a-46b6-b4d6-cbc0f98b1db4)

You can just change the `deployment_template_id` attribute and apply the change through Terraform as well though. 

If there continues to be issues, or you'd like us to look at your deployment specifically then reach out to Support either through the Cloud Console or via support@elastic.co and we can work through the steps.

**Comment by koushikgongireddy** [comment]
Thanks @tobio 
Let me check on this and get back!!
